internal/provider/aws/provider.go

```diff
@@ -91,6 +92,28 @@ func WithNodeName(nodeName string) NodeNameFilter {
 	return NodeNameFilter{NodeName: nodeName}
 }
 
+func isFirewallRuleDuplicate(
+	firewallRules []v1alpha1.FirewallRule,
+	firewallRule *v1alpha1.FirewallRule,
+) bool {
+	frSpec := provider.EncodeFirewallRuleSpec(firewallRule)
+	frSpecs := provider.EncodeFirewallRuleSpecs(firewallRules)
+
+	if frSpec.Direction == provider.DirectionIngress &&
+		slices.ContainsFunc(provider.GetIngressIPPermissions(frSpecs), func(e *provider.IPPermission) bool {
+			return frSpec.Name != firewallRule.Name && reflect.DeepEqual(e, frSpec.IPPermission)
```

Comment:
Are you sure this works ? Since `frSpec.Name` come from `frSpec := provider.EncodeFirewallRuleSpec(firewallRule)`, I think it is alway true ?